src/Microsoft.AspNetCore.Diagnostics.HealthChecks/HealthCheckMiddleware.cs

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.AspNetCore.Diagnostics.HealthChecks
+{
+    public class HealthCheckMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly HealthCheckOptions _healthCheckOptions;
+        private readonly IHealthCheckService _healthCheckService;
+
+        public HealthCheckMiddleware(RequestDelegate next, IOptions<HealthCheckOptions> healthCheckOptions, IHealthCheckService healthCheckService)
+        {
+            _next = next;
+            _healthCheckOptions = healthCheckOptions.Value;
+            _healthCheckService = healthCheckService;
+        }
+
+        /// <summary>
+        /// Process an individual request.
+        /// </summary>
+        /// <param name="context"></param>
+        /// <returns></returns>
+        public async Task InvokeAsync(HttpContext context)
+        {
+            if (context.Request.Path == _healthCheckOptions.Path)
```

Comment:
Should this be startswith? 

What about cases like `/health` vs `/health/` - should this be case-sensitive?

Reply:
I'm legit unsure on this. I definitely thing we want `/health` and `/health/` to work, but I don't think we'd want `/health/foo` to work, since the user may want to map something else to that URL. For now, I'm going to change this to support `/health` and `/health/` and be case-insensitive. We can continue to discuss this though.

Reply:
Hmm, looking at our other middleware, we generally do exact `==` checks for this. For example Authentication handlers for sign-in links: https://github.com/aspnet/Security/blob/dev/src/Microsoft.AspNetCore.Authentication/RemoteAuthenticationHandler.cs#L40

Of course, that's not exactly the same scenario here, since the Auth handlers also generate the urls and can make sure it's correct. Similarly, the Welcome page (not exactly an example of a "useful" middleware, I'll grant) also uses an exact path match:

https://github.com/aspnet/Diagnostics/blob/d390c369ffa7c0d6f83e61a6e01c3f42d8e6c5a0/src/Microsoft.AspNetCore.Diagnostics/WelcomePage/WelcomePageMiddleware.cs#L50

Reply:
== is overriden. https://github.com/aspnet/HttpAbstractions/blob/75a7552360cc1bfed1a56aad2180f730e5ba24cc/src/Microsoft.AspNetCore.Http.Abstractions/PathString.cs#L379

Reply:
I agree that this should be "path equality", so not supporting `/health/foo` is good. Beyond that I don't super care how nitpicky we are about exact matches (casing, trailing slashes, etc.).

Reply:
Agreed... 99% of the time these are called by other services/ scripts anyway so exact matching is neither here nor there. Wild carding the whole path after Healy is a bad idea

Reply:
As far as I can tell, we don't do whatever @Eilon is calling "path equality" (I assume you mean treating `/health` and `/health/` the same) anywhere else in our stack, and the equality operator for `PathString` doesn't do it. If we want to add logic to do that, I can look in to that, but I don't think it's necessary, since as @Drawaes said these are usually just punched into an automated monitoring tool.

Reply:
Cool beans, I just wanted to make sure it was thought through.